### PR TITLE
[CHORE] auto publish attempt 2

### PR DESCRIPTION
**Pull request by spham92**
# Summary
Adding code back revert in #7472 with some differences

- Cron is now scheduled on Wednesday to match emberjs's auto alpha release time
- For finding the next alpha version, I changed the code to just increment the pre-release version in `lerna.json`. This means that we will need a human to bump the minor and major when needed.
- Added `--dryRun` as an option to publish script to skip auto committing and publish when using `lerna` and skipping publishing to npm. This will be useful during local testing

Repo owners need to add `NODE_AUTH_TOKEN` to the secrets for this to fully work.

Use the following command to test what would be the bulk of the GitHub action: `node bin/publish.js canary --dryRun --autoAlphaVersion --force --skipSmokeTest`

# Testing
I tried the GitHub action on a local version of the Ember Data: https://github.com/spham92/data/actions/runs/769888390
It failed because of missing NODE_AUTH_TOKEN which is great since it now throws an error for that

For another instance, I ran a dry run and tested the artifacts: https://github.com/spham92/data/actions/runs/770096237

### CODE REVIEW

**Comment by runspired** [comment]
@spham92 I think I see the assumption made in the first version that was incorrect and I think we could make this a bit safer still.

The semantics of our release cycle is that there is always a stable|beta|canary with different version numbers. E.g. if latest stable is `3.27.x` then the current beta will always be `3.28.0-beta.N` and the current alpha will always be `3.29.0-alpha.N`.

I think since there is potential for human error or for version fixes to past versions to be pushed to github that could appear in front of latest stable (assuming github is sorting chronologically vs by incrementing number) I would prefer that we find the version a different way.

Also considering that for this approach the next canary to be a major/minor requires a manual bump to the version anyway, what prevents us from using the version in package.json?

File: .github/workflows/alpha-release.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 12.x
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Install dependencies for master
+        run: yarn install --frozen-lockfile --non-interactive
+      - name: Make sure git user is setup
+        run: |
+          git config --local user.email 'cron@example.com'
```
Comment (runspired) [inline .github/workflows/alpha-release.yml:44]: reminder to update this. I believe there's a tomster one to use (see ember's config)

File: bin/publish.js [hunk 2]
```diff
@@ -347,28 +370,47 @@ async function main() {
...
+    }
+    // Assume human ran script if token is missing
+    await confirmPublish(tarballs, options, !npmAuthTokenInEnv);
     console.log(`✅ ` + chalk.cyan(`Successfully Published ${nextVersion}`));
   } else {
     console.log('⚠️ ' + chalk.grey(`Skipping Publishing`));
   }
 }
 
+let cli = readline.createInterface({
```
Comment (runspired) [inline bin/publish.js:409]: is this left over from something? I don't see it used in this PR

**Review by runspired — APPROVED** [review]
Overall 👍🏻 . Wouldn't mind @rwjblue giving it a pass. A couple of minor cleanups to do before we merge.

Also @rwjblue or @igorT will need to setup the secret as I cannot.

File: bin/publish.js (hunk 2 above)
Comment (spham92) [reply bin/publish.js:409]: good catch, reverted

File: .github/workflows/alpha-release.yml (hunk 1 above)
Comment (spham92) [reply .github/workflows/alpha-release.yml]: updated to:

```
      - name: Make sure git user is setup
        run: |
          git config --local user.email 'tomster@emberjs.com'
          git config --local user.name 'Ember.js Alpha Releaser'
```

**Review by runspired — APPROVED** [review]

**Pull request merged**